tests/forms_tests/tests/test_media.py

```diff
@@ -571,3 +581,33 @@ def test_merge_css_three_way(self):
         # was never specified.
         merged = widget3 + form1 + form2
         self.assertEqual(merged._css, {'screen': ['a.css', 'b.css'], 'all': ['c.css']})
+
+    def test_merge_js_three_way_non_pairwise(self):
```

Comment:
I don't know if you need to actually create all the Medias. At this point a unit tests calling merge is sufficient.
What I would like to see is:
```python
assert list(merge([1,2],[1,3],[2,3], [5,7], [5,6], [6,7,9], [8,9])) == [1,2,3,5,6,7,8,9]
```

Reply:
Hmm that's right. I copied the tests from @gasman 's PR because they perfectly describe the bug which is fixed by this PR in a way which communicates the problem. While it's certainly possible to reduce the test suite I sort of like the verbosity since it's more obvious what should happen and what was happening instead. 

But that's just my preference -- but if you don't insist I'd like to wait for more feedback or opinions on this point, OK?

Reply:
Addendum: More low-level tests could also be added to `test_topological_sort.py`.